### PR TITLE
Try Reason Master Branch For new ppx-let syntax.

### PR DESCRIPTION
**Pull request by jordwalke**
Summary:I wanted to see how the codebase looks with @let-def's new ppx
extension sugar printing. Not bad. Some first class syntax for let%bind
will look even better, but this is definitely tolerable.
There is some suboptimal printing of `let%bind` which I'll point out in this PR comments. If you want to hold off on merging this because of those suboptimal printings, I understand.

Test Plan:

Reviewers:

CC:

### CODE REVIEW

File: lib/Builder.re [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  let prepare = () => {
+    let%bind () = mkdir(installDir);
+    let%bind () = mkdir(stageDir);
+    let%bind () =
+      if (spec.sourceType == Immutable || spec.buildType == InSource) {
+        rmdir(buildDir);
+      } else {
+        ok;
+      };
+    let%bind () = mkdir(buildDir);
```
Comment (jordwalke) [inline lib/Builder.re:131]: You should have seen how absurd the printing was before @let-def's diff! 😆

File: lib/Config.re [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      "/"
+      ++ storeVersion
+      ++ "/"
+      ++ storeInstallTree
+      ++ "/"
+      ++ ocamlrunStorePath
+    );
+};
+
+/**
```
Comment (jordwalke) [inline lib/Config.re:35]: Down below:
This is the main suboptimal printing that only occurs with `ppx` extension point sugar.
Normally, local opens of this form would be printed as:

```reason
{
  open Run;
  let%bind prefixPath =
    switch prefixPath {
    | Some(v) => Ok(v)
    | None =>
      let%bind home = Bos.OS.Dir.user();
      Ok(home / ".esy");
    };
}
```

We just need to special case that same rule for ppx extension point sugar.

Comment (andreypopp) [reply lib/Config.re:35]: Heh, the same bug was with my fork — if `let ... = do ...` was the first expr in a sequence then the local open was printed like that.

**Comment by andreypopp** [comment]
Thanks!

**Pull request merged**